core/eolearn/core/utilities/other.py

```diff
@@ -0,0 +1,199 @@
+"""
+The utilities module is a collection of classes and functions used across the eolearn package, such as checking whether
+two objects are deeply equal, padding of an image, etc.
+
+Credits:
+Copyright (c) 2017-2022 Matej Aleksandrov, Matej Batič, Grega Milčinski, Domagoj Korais, Matic Lubej (Sinergise)
+Copyright (c) 2017-2022 Žiga Lukšič, Devis Peressutti, Tomislav Slijepčević, Nejc Vesel, Jovan Višnjić (Sinergise)
+Copyright (c) 2017-2022 Anže Zupanc (Sinergise)
+Copyright (c) 2019-2020 Jernej Puc, Lojze Žust (Sinergise)
+Copyright (c) 2017-2019 Blaž Sovdat, Andrej Burja (Sinergise)
+
+This source code is licensed under the MIT license found in the LICENSE
+file in the root directory of this source tree.
+"""
+
+import logging
+import warnings
+from logging import Filter, LogRecord
+from typing import Optional
+
+import uuid
+import numpy as np
+import geopandas as gpd
+from geopandas.testing import assert_geodataframe_equal
+
+from ..exceptions import EODeprecationWarning
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class LogFileFilter(Filter):
```

Comment:
Perhaps this class could be in a separate utils module `logging.py`? It is very specific compared to other things in this module and I imagine that in the future we might have more logging utils.